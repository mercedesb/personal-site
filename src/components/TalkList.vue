<template>
  <div v-if="talks.length" class="TalkList">
    <smart-link :class="'TalkListItem TalkListItem--' + talk.color + ' ' + talk.iconCssClass" v-for="talk in talks" v-bind="talk" :key="talk.id" :to="'/speaking/' + talk.urlSegment">
      <div class="TalkListItem-icon" v-html="talk.iconSvg"></div>
      <h3 class="TalkListItem-title">{{talk.title}}</h3>
      <div class="TalkListItem-moreInfo">
        Resources
      </div>
    </smart-link>
  </div>
</template>

<script>
export default {
  props: {
    talks: Array 
  }
}
</script>

<style lang="scss">
@import '../assets/styles/variables.scss';

.TalkList {
  display: flex;
  flex-wrap: wrap;
  max-width: 1000px;
  margin: 0 auto;
}

.TalkListItem {
  width: 100%;
  margin: $base-spacing $small-spacing;

  @include media($min-tablet) {
    width: 45%;
  }

  @include media($min-desktop) {
    width: 30%;
  }

  > div, > a {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: column;
  }

  &-icon {
    width: 30%;
    margin: $base-spacing 0 $small-spacing;

    @include light-svg;

  }

  &-title {
    font-size: $large-font-size;
    text-align: center;
    color: $white;
    margin: $small-spacing;
  }

  &-moreInfo {
    opacity: 0;
    transition: opacity .5s;
    font-weight: $heavy-font-weight;
    text-decoration: underline;
    color: $white;
    margin: $small-spacing 0 $base-spacing;
  }

  &:hover {
    cursor: pointer;

    .TalkListItem-moreInfo {
      opacity: 1;
      transition: opacity .5s;
    }
  }

  @include background-color;
  @include hover-saturate;

}

</style>
