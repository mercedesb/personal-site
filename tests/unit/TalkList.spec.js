import TalkList from "@/components/TalkList.vue";
import { TestUtility } from "./TestUtility";

describe("TalkList", () => {
  let component;

  const initialProps = {
    talks: TestUtility.talks
  };

  const shallow = propsData =>
    TestUtility.shallow(TalkList, {
      propsData: {
        ...initialProps,
        ...propsData
      }
    });

  describe("Snapshots", () => {
    describe("with talks", () => {
      it("matches snapshot", () => {
        component = shallow();
        expect(component.element).toMatchSnapshot();
      });
    });

    describe("without talks", () => {
      it("matches snapshot", () => {
        component = shallow({ talks: [] });
        expect(component.element).toMatchSnapshot();
      });
    });
  });
});
