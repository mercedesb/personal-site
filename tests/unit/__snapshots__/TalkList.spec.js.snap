// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkList Snapshots matches snapshot 1`] = `
<div
  class="TalkList"
>
  <smart-link-stub
    class="TalkListItem TalkListItem--gray Draw--300"
    color="gray"
    givenat="[object Object]"
    iconcssclass="Draw--300"
    iconsvg=<svg></svg>
    id="talk2"
    maincontent="talk main content"
    preamble="talk 2 preamble"
    title="talk 2 title"
    to="speaking/talk2"
    urlsegment="talk2"
  >
    <div
      class="TalkListItem-icon"
    >
      <svg />
    </div>
     
    <h3
      class="TalkListItem-title"
    >
      talk 2 title
    </h3>
     
    <div
      class="TalkListItem-moreInfo"
    >
      
      Resources
    
    </div>
  </smart-link-stub>
  <smart-link-stub
    class="TalkListItem TalkListItem--gray Draw--300"
    color="gray"
    givenat="[object Object]"
    iconcssclass="Draw--300"
    iconsvg=<svg></svg>
    id="talk1"
    maincontent="talk main content"
    preamble="talk 1 preamble"
    title="talk 1 title"
    to="speaking/talk1"
    urlsegment="talk1"
  >
    <div
      class="TalkListItem-icon"
    >
      <svg />
    </div>
     
    <h3
      class="TalkListItem-title"
    >
      talk 1 title
    </h3>
     
    <div
      class="TalkListItem-moreInfo"
    >
      
      Resources
    
    </div>
  </smart-link-stub>
  <smart-link-stub
    class="TalkListItem TalkListItem--gray Draw--300"
    color="gray"
    givenat="[object Object]"
    iconcssclass="Draw--300"
    iconsvg=<svg></svg>
    id="talk3"
    maincontent="talk main content"
    preamble="talk 3 preamble"
    title="talk 3 title"
    to="speaking/undefined"
  >
    <div
      class="TalkListItem-icon"
    >
      <svg />
    </div>
     
    <h3
      class="TalkListItem-title"
    >
      talk 3 title
    </h3>
     
    <div
      class="TalkListItem-moreInfo"
    >
      
      Resources
    
    </div>
  </smart-link-stub>
</div>
`;
